contrib/natss/pkg/dispatcher/channel/controller.go

```diff
@@ -56,5 +56,14 @@ func ProvideController(ss *dispatcher.SubscriptionsSupervisor, mgr manager.Manag
 		return nil, err
 	}
 
+	// We need to force reconciliation when the NATS client disconnects.
+	err = c.Watch(&source.Channel{
```

Comment:
line 51 already adds watch to channel, do we need both?

Reply:
Updated with a more verbose and hopefully clearer comment.